apis/AppGateway/IAppGateway.h

```diff
@@ -0,0 +1,193 @@
+            // @brief Check if the application belongs to a given permission group
+            virtual Core::hresult CheckPermissionGroup(const string& appId ,
+                                                       const string& permissionGroup ,
+                                                       bool& allowed /* @out */) = 0;
+
+        };
+
+        // @text:keep
+        struct EXTERNAL IAppGatewayResponder : virtual public Core::IUnknown
+        {
+            enum
+            {
+                ID = ID_APP_GATEWAY_RESPONDER
+            };
+
+            // @json:omit
+            // @text respond
+            // @brief Provides support for responding to a given context
+            // @param context: Execution context containing requestId, connectionId, appId
+            // @param payload: the response payload
+            // @returns Core::hresult
+            virtual Core::hresult Respond(const GatewayContext& context ,
+                                          const string& payload /*@opaque */) = 0;
+
+            // @json:omit
+            // @text Emit
+            // @brief Provides support for Emitting Notifications to a given context
+            // @param context: Execution context containing requestId, connectionId, appId
+            // @param payload: the response payload
+            // @returns Core::hresult
+            virtual Core::hresult Emit(const GatewayContext& context ,
+                const string& method , const string& payload /*@opaque */) = 0;
+
+            // @json:omit
+            // @text Request
+            // @brief Forwards a Request to the Client. Needed for App Provider Patterns.
+            // @param connectionId: Connection Id
+            // @param id: Request id
+            // @param method: Method
+            // @param params: Params string object
+            // @returns Core::hresult
+            virtual Core::hresult Request(const uint32_t connectionId , 
+                const uint32_t id , const string& method , const string& params /*@opaque */) = 0;
+
+            
+            // @json:omit
+            // @text GetGatewayConnectionContext
+            // @brief Gets any connection context parameter like headers, url params
+            // @param connectionId: Connection Id
+            // @param contextKey: Connection Id
+            // @param contextValue: response value
+            // @returns Core::hresult
+            virtual Core::hresult GetGatewayConnectionContext(const uint32_t connectionId ,
+                const string& contextKey ,
+                string& contextValue /* @out */) = 0;
+
+
+            struct EXTERNAL INotification : virtual public Core::IUnknown
+            {
+                enum { ID = ID_APP_GATEWAY_CONNECTION_NOTIFICATION };
+
+                // @brief Notifies App has either started or stopped a connection. App can create multiple connections.
+                // @text OnAppConnectionChanged
```

Comment:
Same as previous comment. Please change this to camel case (onAppConnectionChanged).